cmds/core/netcat/netcat.go

```diff
@@ -16,59 +16,106 @@ import (
 	"github.com/u-root/u-root/pkg/uroot/util"
 )
 
-const usage = "netcat [go-style network address]"
+const usage = "netcat host port"
 
-var (
-	netType = flag.String("net", "tcp", "What net type to use, e.g. tcp, unix, etc.")
-	listen  = flag.Bool("l", false, "Listen for connections.")
-	verbose = flag.Bool("v", false, "Verbose output.")
-)
+var errMissingHostnameOrPort = fmt.Errorf("missing hostname or port")
+
+type params struct {
+	network string
+	listen  bool
+	verbose bool
+	host    string
+	port    string
+}
+
+func parseParams() params {
+	netType := flag.String("net", "tcp", "What net type to use, e.g. tcp, unix, etc.")
+	listen := flag.Bool("l", false, "Listen for connections.")
+	verbose := flag.Bool("v", false, "Verbose output.")
+	flag.Parse()
+
+	return params{
```

Comment:
I am wondering if params should not just include the host and port too?

Reply:
They can, but in find it easier to do error handling when parseParams works only with flag arguments and deal with non flag arguments later. But I can change that if needed.